adadamp/_dist.py

```diff
@@ -71,18 +84,33 @@ def gradient(
     where `l` is the loss function for a single example.
 
     """
-    model, _ = model_opt
+    
+    # Workaround: Gradients should be cleared when entering this funciton, 
+    #     but at this moment this behavior is not occuring
+    old_model, _ = model_opt
+    model = deepcopy(old_model)
```

Comment:
How much time does this line take? I'd look at the Dask dashboard for this https://distributed.dask.org/en/latest/diagnosing-performance.html#statistical-profiling